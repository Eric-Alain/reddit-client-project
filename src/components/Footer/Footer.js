//React
import * as React from 'react'

//3rd party
import PropTypes from 'prop-types'
import Col from 'react-bootstrap/Col'
import Container from 'react-bootstrap/Container'
import Row from 'react-bootstrap/Row'

const Footer = ({ author, siteUrl }) => (
  <footer>
    <Container className='px-sm-0 rounded'>
      <Row>
        <Col xs='6'>
          <p className='mb-0 py-3 lh-1'>
            Original solution by: <a href={siteUrl}>{author}</a>
          </p>
        </Col>
        <Col xs='6' className='text-end'>
          <p className='mb-0 py-3 lh-1'>
            Challenge project by:{' '}
            <a href='https://www.codecademy.com/paths/front-end-engineer-career-path/tracks/fecp-22-portfolio-project-reddit-client/modules/wdcp-22-reddit-client/kanban_projects/reddit-client'>
              Codecademy.com
            </a>
          </p>
        </Col>
      </Row>
    </Container>
  </footer>
)

Footer.propTypes = {
  author: PropTypes.string.isRequired,
  siteUrl: PropTypes.string.isRequired
}

export default Footer
