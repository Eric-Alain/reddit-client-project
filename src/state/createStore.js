//Redux
import { createStore as reduxCreateStore, combineReducers } from 'redux'
import { composeWithDevTools } from 'redux-devtools-extension'

//Reducers
import searchReducer from '../state/reducers/search'
import dataReducer from '../state/reducers/data'
import subredditsReducer from '../state/reducers/subreddits'
import togglesReducer from '../state/reducers/toggles'

//Create root reducer
export const rootReducer = combineReducers({
  search: searchReducer,
  data: dataReducer,
  subreddits: subredditsReducer,
  toggles: togglesReducer
})

//Create store, pass preloadedState via plugin
const createStore = preloadedState => {
  return reduxCreateStore(rootReducer, preloadedState, composeWithDevTools())
}
export default createStore
