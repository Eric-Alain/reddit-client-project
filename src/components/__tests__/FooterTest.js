import React from 'react'
import renderer from 'react-test-renderer'
import Footer from '../Footer/Footer'

import { Provider } from 'react-redux'
import createStore from '../../state/createStore'
import registerIcons from '../../icons/icons'

const store = createStore()
registerIcons()

describe('Footer', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Footer author='Eric Alain' siteUrl='https://www.ericalain.ca' />
        </Provider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
