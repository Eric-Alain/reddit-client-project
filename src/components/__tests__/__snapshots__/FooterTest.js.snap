// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<footer>
  <div
    className="px-sm-0 rounded container"
  >
    <div
      className="row"
    >
      <div
        className="col-6"
      >
        <p
          className="mb-0 py-3 lh-1"
        >
          Original solution by: 
          <a
            href="https://www.ericalain.ca"
          >
            Eric Alain
          </a>
        </p>
      </div>
      <div
        className="text-end col-6"
      >
        <p
          className="mb-0 py-3 lh-1"
        >
          Challenge project by:
           
          <a
            href="https://www.codecademy.com/paths/front-end-engineer-career-path/tracks/fecp-22-portfolio-project-reddit-client/modules/wdcp-22-reddit-client/kanban_projects/reddit-client"
          >
            Codecademy.com
          </a>
        </p>
      </div>
    </div>
  </div>
</footer>
`;
